cylc/flow/task_pool.py

```diff
@@ -686,6 +686,7 @@ def remove(self, itask, reason=""):
             if not self.hidden_pool[itask.point]:
                 del self.hidden_pool[itask.point]
             LOG.debug(f"[{itask}] {msg}")
+            self.task_queue_mgr.remove_task(itask)
```

Comment:
Unrelated bug?

Reply:
It is a bug fix, but not unrelated. Expired tasks need to be removed as complete (possibly still subject to debate!) but the "remove" process was not removing the target task from the queues.

Reply:
Having run into this bug again, I'll cherry-pick it to a separate bug fix PR. (As task expiry is a bigger discussion).

Reply:
Having run into this bug again, I'll cherry-pick it to a separate bug fix PR. (As task expiry is a bigger discussion).... #5573

Reply:
... #5573